pylsp_ruff/plugin.py

```diff
@@ -487,6 +487,24 @@ def run_ruff_format(
     )
 
 
+@lru_cache
+def find_executable(executable):
+    if executable is not None:
+        cmd = [executable]
+        try:
+            Popen(cmd, stdin=PIPE, stdout=PIPE, stderr=PIPE).communicate()
+        except Exception:
+            log.error(f"Can't execute ruff with given executable '{executable}'.")
+        finally:
+            return cmd
+    else:
+        cmd = [sys.executable, "-m", "ruff"]
+        (_, stderr) = Popen(cmd, stdin=PIPE, stdout=PIPE, stderr=PIPE).communicate()
+        if "No module named ruff" in stderr.decode():
+            cmd = [shutil.which("ruff")]
```

Comment:
this seems rather hacky, especially just looking at the output, which i assume is not stable for that error message :)

you could instead try to import the ruff module (try: import ruff except ImportError), or as mentioned before check if it exists as entry point.